File: app/src/main/java/mm/kso/movieapplication/viewmodels/ActorDetailslViewModel.java
package mm.kso.movieapplication.viewmodels;

import android.util.Log;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.HashMap;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;
import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
import io.reactivex.rxjava3.disposables.CompositeDisposable;
import io.reactivex.rxjava3.schedulers.Schedulers;
import mm.kso.movieapplication.model.Actor;
import mm.kso.movieapplication.repository.Repository;

@HiltViewModel
public class ActorDetailslViewModel extends ViewModel {

    private static final String TAG = "HomeViewModel";
    private Repository repository;
    private final CompositeDisposable disposable = new CompositeDisposable();
    private MutableLiveData<Actor> actorDetails = new MutableLiveData<>();

    @Inject
    public ActorDetailslViewModel(Repository repository) {
        this.repository = repository;
    }

    public MutableLiveData<Actor> getActor() {
        return actorDetails;
    }

    public void getActorDetails(int personId, HashMap<String,String> map) {
        disposable.add(repository.getActorDetails(personId, map)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(result -> actorDetails.setValue(result),
                        error -> Log.e(TAG, "getActorDetails: " + error.getMessage()))
        );
    }

}

File: app/src/main/java/mm/kso/movieapplication/viewmodels/ActorDetailslViewModel.kt
package mm.kso.movieapplication.viewmodels

import android.util.Log
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import dagger.hilt.android.lifecycle.HiltViewModel
import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
import io.reactivex.rxjava3.disposables.CompositeDisposable
import io.reactivex.rxjava3.schedulers.Schedulers
import mm.kso.movieapplication.model.Actor
import mm.kso.movieapplication.repository.Repository
import java.util.*
import javax.inject.Inject

@HiltViewModel
class ActorDetailslViewModel @Inject constructor(private val repository: Repository) : ViewModel() {
    private val disposable = CompositeDisposable()
    val actor = MutableLiveData<Actor>()
    fun getActorDetails(personId: Int, map: HashMap<String?, String?>?) {
        disposable.add(
            repository.getActorDetails(personId, map)
                ?.subscribeOn(Schedulers.io())
                ?.observeOn(AndroidSchedulers.mainThread())
                ?.subscribe(
                    { result: Actor? -> actor.setValue(result) }
                ) { error: Throwable -> Log.e(TAG, "getActorDetails: " + error.message) }
        )
    }

    companion object {
        private const val TAG = "HomeViewModel"
    }
}